packages/contracts-sdk/src/lib/addresses.ts

```diff
@@ -22,6 +22,164 @@ export interface TokenInfo {
   isNewPKP: boolean;
 }
 
+/**
+ * Converts a public key between compressed and uncompressed formats.
+ *
+ * @param publicKey - Public key as a Buffer (33 bytes compressed or 65 bytes uncompressed)
+ * @param compressed - Boolean flag indicating whether the output should be compressed
+ * @returns Converted public key as a Buffer
+ */
+export function publicKeyConvert(
```

Comment:
Do we need to export this method? Seems like an internal util

Reply:
Yes, it is also used in [pkp-base](https://github.com/LIT-Protocol/js-sdk/pull/622/files#diff-d5bbc3deae289e4022c8b5430b1b455cdb2e4c887da2ce9f39e61a4ef2d6ba4c)

Reply:
Do you think it makes sense to move these to `misc` as we already import that package in both `contracts-sdk` and `pkp-base`

Reply:
My main concern is introducing `contracts-sdk` as a dependency to `pkp-base` currently these are the main imports from that package


```
import { publicKeyConvert } from 'secp256k1';

import { LitNodeClient } from '@lit-protocol/lit-node-client';
import { logError } from '@lit-protocol/misc';
import {
  AuthenticationProps,
  JsonExecutionSdkParams,
  PKPBaseProp,
  AuthSig,
  PKPBaseDefaultParams,
  SigResponse,
  RPCUrls,
  AuthMethod,
  SessionSigsMap,
} from '@lit-protocol/types';
``` 

I am wondering if we can strip all deps from these new key converters that are not part of the runtime api. such that they can be migrated without bringing over dependencies. I am a bit worried about making `contracts-sdk` a required dep as it will be rather bloated with the ABI package requirement

Reply:
The import path was already `pkp-base` -> `lit-node-client` -> `lit-node-client-nodejs` -> `core` -> `contracts-sdk` so this is not really creating the link between the two

You mean implementing `bech32`, `bs58`, etc. ourselves?

Reply:
Sorry I should have mentioned that I was assuming the refactor of `core` to no longer depend on `contracts-sdk`
Ideally, as we want to break the coupling of `core` and `contracts-sdk`  since we really only use it to resolve the staking contract. If we bind to it here for these utilities we end up kind of back at a similar points.

Reply:
Oh got it now, considering that makes sense. Moving...